Mathlib/Topology/Meagre.lean

```diff
@@ -0,0 +1,148 @@
+/-
+Copyright (c) 2023 Michael Rothgang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Michael Rothgang
+-/
+import Mathlib.Topology.GDelta
+
+/-!
+## Nowhere dense and meagre sets
+Define nowhere dense and meagre sets (`IsNowhereDense` and `IsMeagre`, respectively)
+and prove their basic properties.
+
+Main properties:
+- `closure_nowhere_dense`: the closure of a nowhere dense set is nowhere dense
+- `closed_nowhere_dense_iff_complement`: a closed set is nowhere dense iff
+its complement is open and dense
+- `meagre_iff_complement_comeagre`: a set is nowhere dense iff its complement is `residual`.
+- `empty_mono`: subsets of meagre sets are meagre
+- `meagre_iUnion`: countable unions of meagre sets are meagre
+-/
+
+open Function TopologicalSpace Set
+set_option autoImplicit false
+
+variable {α : Type*} [TopologicalSpace α]
+
+/-- A set is nowhere dense iff its closure has empty interior. -/
+def IsNowhereDense (s : Set α) := IsEmpty (interior (closure s))
```

Comment:
`IsEmpty` involves an extra coercion to type. You should be able to write simply `interior (closure s) = ∅`.